okta/table_okta_group_rule.go

```diff
@@ -0,0 +1,141 @@
+package okta
+
+import (
+	"context"
+	"strings"
+
+	"github.com/okta/okta-sdk-golang/v2/okta"
+	"github.com/okta/okta-sdk-golang/v2/okta/query"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+func tableOktaGroupRule() *plugin.Table {
+	return &plugin.Table{
+		Name:        "okta_group_rule",
+		Description: "Retrieve group rules for Okta. Group rules define conditions and actions for automating group membership.",
+		Get: &plugin.GetConfig{
+			Hydrate:           getOktaGroupRule,
+			KeyColumns:        plugin.SingleColumn("id"),
+			ShouldIgnoreError: isNotFoundError([]string{"Not found"}),
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listOktaGroupRules,
+		},
+		Columns: []*plugin.Column{
+			// Basic columns
+			{Name: "id", Type: proto.ColumnType_STRING, Description: "Unique identifier of the group rule."},
+			{Name: "name", Type: proto.ColumnType_STRING, Description: "Name of the group rule."},
+			{Name: "status", Type: proto.ColumnType_STRING, Description: "Status of the group rule (e.g., ACTIVE, INACTIVE)."},
+			{Name: "created", Type: proto.ColumnType_TIMESTAMP, Description: "Timestamp when the group rule was created."},
+			{Name: "last_updated", Type: proto.ColumnType_TIMESTAMP, Description: "Timestamp when the group rule was last updated."},
+
+			// JSON columns
+			{Name: "conditions", Type: proto.ColumnType_JSON, Description: "Conditions that trigger this group rule."},
+			{Name: "actions", Type: proto.ColumnType_JSON, Description: "Actions performed when the rule conditions are met."},
+
+			// Steampipe-specific
+			{Name: "title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Name"), Description: "Title of the group rule."},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listOktaGroupRules(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	client, err := Connect(ctx, d)
+	if err != nil {
+		logger.Error("listOktaGroupRules", "connection_error", err)
+		return nil, err
+	}
+
+	// Default maximum limit set as per documentation
+	// https://developer.okta.com/docs/reference/api/groups/#list-group-rules
+	input := query.Params{
+		Limit: 200,
+	}
+
+	// If the requested number of items is less than the paging max limit
+	// set the limit to that instead
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < input.Limit {
+			input.Limit = *limit
+		}
+	}
+
+	// Fetch group rules
+	groupRules, resp, err := client.Group.ListGroupRules(ctx, &input)
+	if err != nil {
+		logger.Error("listOktaGroupRules", "list_group_rules_error", err)
+		if strings.Contains(err.Error(), "Not found") {
```

Comment:
Can we add the IgnoreConfig [here](https://github.com/turbot/steampipe-plugin-okta/pull/150/files#diff-da271dc5a376e5e5b97854e3db5bd3d02c3e3e43140c269cfd9503934c1fa3ceR24) instead of handling it manually?